### PR TITLE
[FEATURE] Enhance hook provider with methods passing on the modified value

### PR DESCRIPTION
**Pull request by mk-mxp**
This is required for the next hooks that modify method scoped values.

### CODE REVIEW

File: Classes/Hooks/HookProvider.php [hunk 1]
```diff
@@ -123,6 +124,26 @@ public function executeHookReturningMergedArray(string $method, ...$params): arr
...
+    /**
+     * Executes the hooked-in methods that pass on a manipulated value.
+     *
+     * @param string $method the method to execute
+     * @param mixed $value the value to manipulate by `$method()`
+     * @param mixed $params parameters to `$method()`
+     *
+     * @return mixed the manipulated value
+     */
+    public function executeHookReturningModifiedValue(string $method, $value, ...$params)
```
Comment (oliverklee) [inline Classes/Hooks/HookProvider.php]: Please don't modify or overwrite arguments in a method. Instead, please copy its value into a new local variable. (This makes reading the method easier as the reader usually assumes that `$value` is the parameter, not a new variable.)

File: Tests/Unit/Hooks/Fixtures/TestingHookImplementorReturnsModifiedValue.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+final class TestingHookImplementorReturnsModifiedValue implements TestingHookInterfaceReturnsModifiedValue
+{
+    /**
+     * Gets called during HookProvider tests.
+     *
+     * @return bool
+     */
+    public function testHookMethodReturnsModifiedBool(bool $value): bool
+    {
+        return $value === false ? $value : !$value;
```
Comment (oliverklee) [inline Tests/Unit/Hooks/Fixtures/TestingHookImplementorReturnsModifiedValue.php]: Isn't this effectively the same as `return false;`?

File: Tests/Unit/Hooks/Fixtures/TestingHookImplementorReturnsModifiedValue.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    /**
+     * Gets called during HookProvider tests.
+     *
+     * @return int
+     */
+    public function testHookMethodReturnsModifiedInt(int $value): int
+    {
+        return $value < 0 ? $value : $value + 1;
```
Comment (oliverklee) [inline Tests/Unit/Hooks/Fixtures/TestingHookImplementorReturnsModifiedValue.php]: Why not use a simpler  `return $value +1;` (or something similar)?

File: Tests/Unit/Hooks/Fixtures/TestingHookInterfaceReturnsModifiedValue.php [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * @author Michael Kramer <m.kramer@mxp.de>
+ */
+interface TestingHookInterfaceReturnsModifiedValue extends Hook
+{
+    /**
+     * Gets called during HookProvider tests.
+     *
+     * @return bool
+     */
+    public function testHookMethodReturnsModifiedBool(bool $value): bool;
```
Comment (oliverklee) [inline Tests/Unit/Hooks/Fixtures/TestingHookInterfaceReturnsModifiedValue.php]: For all the interfaces, please either drop all `@param` and `@return` that do not add anything more than the type declarations say, or please have them all. (Here, only `@return` is present, but not `@param`.)

File: Tests/Unit/Hooks/Fixtures/TestingHookImplementorReturnsModifiedValue.php (hunk 3 above)
Comment (mk-mxp) [reply Tests/Unit/Hooks/Fixtures/TestingHookImplementorReturnsModifiedValue.php]: I didn't see that, thanks. I was coming from the idea of modifying `$value` alsways...

File: Tests/Unit/Hooks/Fixtures/TestingHookImplementorReturnsModifiedValue.php (hunk 2 above)
Comment (mk-mxp) [reply Tests/Unit/Hooks/Fixtures/TestingHookImplementorReturnsModifiedValue.php]: I didn't see that, thanks. I was coming from the idea of modifying `$value` alsways...

File: Tests/Unit/Hooks/Fixtures/TestingHookInterfaceReturnsModifiedValue.php (hunk 4 above)
Comment (mk-mxp) [reply Tests/Unit/Hooks/Fixtures/TestingHookInterfaceReturnsModifiedValue.php]: Sorry, didn't see that, too.

**Pull request merged**